src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala

```diff
@@ -99,19 +100,22 @@ class PendingTransactionsManager(txPoolConfig: TxPoolConfig, peerManager: ActorR
       val timestamp = System.currentTimeMillis()
       pendingTransactions.put(newStx.hash, PendingTransaction(newStx, timestamp))
 
-      (peerManager ? PeerManagerActor.GetPeers).mapTo[Peers].foreach {
-        peers => peers.handshaked.foreach { peer => self ! NotifyPeer(List(newStx), peer) }
-      }
+      (peerManager ? PeerManagerActor.GetPeers).mapTo[Peers]
+        .map(_.handshaked)
+        .filter(_.nonEmpty)
+        .foreach { peers => self ! NotifyPeers(Seq(newStx), peers) }
 
-    case NotifyPeer(signedTransactions, peer) =>
+    case NotifyPeers(signedTransactions, peers) =>
       pendingTransactions.cleanUp()
-      val txsToNotify = signedTransactions
-        .filter(stx => pendingTransactions.asMap().containsKey(stx.hash)) // signed transactions that are still pending
-        .filterNot(isTxKnown(_, peer.id)) // and not known by peer
-
-      if (txsToNotify.nonEmpty) {
-        etcPeerManager ! EtcPeerManagerActor.SendMessage(SignedTransactions(txsToNotify), peer.id)
-        txsToNotify.foreach(setTxKnown(_, peer.id))
+      peers.foreach{
+        peer => val txsToNotify = signedTransactions
+          .filter(stx => pendingTransactions.asMap().containsKey(stx.hash)) // signed transactions that are still pending
+          .filterNot(isTxKnown(_, peer.id)) // and not known by peer
```

Comment:
as we are doing improvments here, maybe we would move `isTxKnown(_, peer.id)` to filter above to have something like `.filter(stx => pendingTransactions.asMap().containsKey(stx.hash) && !isTxKnown(_, peer.id))` to avoid creating one unecessery collection ?

Reply:
nvm :D